### DISCUSSION TITLE
Possible to merge arrays using their values as keys?

### DISCUSSION THREAD

**Discussion by kyguy** [mikefarah/yq#791] [Q&A] [answered]
Is it possible to merge arrays using values as keys?

`x1.yaml`
```
x:
  - a
  - b
```
`x2.yaml`
```
x:
  - b
  - c
```
We can merge arrays using their indices as keys:
```
yq ea 'select(fi==0) *d select(fi==1)' x1.yaml x2.yaml
```
```
x:
  - b
  - c
```
But is there a way to merge arrays using their values as keys in order to remove duplicate values?
```
x:
  - a
  - b
  - c
```

**Comment by mikefarah** [comment, marked as answer]
You should be able to do it - I've recently added an example of merging arrays using matching object keys:

https://mikefarah.gitbook.io/yq/operators/multiply-merge#merge-arrays-of-objects-together-matching-on-a-key

**Reply by mikefarah** [reply]
Looks like all you'd need to do is drop the '.a' bit in the select (so i think it would be select(. == $cur)

**Reply by kyguy** [reply]
Thanks for the example, I appreciate it! 

I tried dropping the '.a' in the select statement
```
yq eval-all '(select(fi==1) | .[]) as $two | select(fi==0) | .[] |= (. as $cur |  $cur * ($two | select(. == $cur)))' x1.yaml x2.yaml
```
and got the following output:
```
x:
  - a
  - b
```
Since we are working with array values I think we need to splat the input we pass to the select statement, but I am still trying to find the right combination to get the select statement working on the array values. Once I find it l will post it here. Thanks again for your help!

**Reply by mikefarah** [reply]
In v4.9.0+ I added a `unique` operator so you can:

```
yq ea 'select(fi==0).x = (select(fi==0).x + select(fi==1).x | unique) | select(fi==0)' data1.yml data2.yml
```

**Reply by kyguy** [reply]
Wow, this is awesome! I was having trouble getting it right before but this new feature makes it super easy! Thank you so much for your help!
